src/core/namespace.c

```diff
@@ -168,16 +167,21 @@ static int append_mounts(BindMount **p, char **strv, MountMode mode) {
 
         STRV_FOREACH(i, strv) {
                 bool ignore = false;
+                char *path;
 
                 if (IN_SET(mode, INACCESSIBLE, READONLY, READWRITE) && startswith(*i, "-")) {
                         (*i)++;
                         ignore = true;
                 }
 
-                if (!path_is_absolute(*i))
+                path = strdup(*i);
+                if (!path)
+                        return -ENOMEM;
+
+                if (!path_is_absolute(path))
```

Comment:
Hey @evverx could you please check again, since I reworked the whole logic and those paths are cleaned https://github.com/systemd/systemd/pull/4594/files#diff-f315b72505d5f0a92ec2f4d068d4f916R913 here and in different places where we chase symlinks, drop duplicate paths etc...